### PR TITLE
Adding doc for hub cluster template support

### PR DESCRIPTION
**Pull request by dockerymick**
https://github.com/open-cluster-management/backlog/issues/14953

### CODE REVIEW

File: governance/custom_template.adoc [hunk 1]
```diff
@@ -1,7 +1,13 @@
 [#support-templates-in-config-policies]
 = Support for templates in configuration policies
 
-Configuration policies support the inclusion of Golang text templates in the object definitions. These templates are resolved at runtime on the target managed cluster using configurations related to that cluster. This gives you the ability to define configuration policies with dynamic content, and inform or enforce Kubernetes resources that are customized to the target cluster.
+Configuration policies support the inclusion of Golang text templates in the object definitions. These templates are resolved at runtime on the target managed cluster using configurations related to that cluster. This gives you the ability to define configuration policies with dynamic content, and inform or enforce Kubernetes resources that are customized to the target cluster. {product-title} supports both hub cluster and managed cluster templates.
```
Comment (mprahl) [inline governance/custom_template.adoc]:
```suggestion
Configuration policies support the inclusion of Golang text templates in the object definitions. These templates are resolved at runtime either on the hub cluster or the target managed cluster using configurations related to that cluster. This gives you the ability to define configuration policies with dynamic content, and inform or enforce Kubernetes resources that are customized to the target cluster.
```

File: governance/custom_template.adoc [hunk 2]
```diff
@@ -361,3 +366,58 @@ spec:
         {{ (fromConfigMap "site-config" "site1" "enabled")  | toBool }}
 ----
 
+[#hub-templates]
+== Support for hub cluster templates in configuration policies
+
+{product-title-short} also supports hub cluster templates to define configuration policies that are dynamically customized to the target cluster. This prevents the creation of separate policies for each target cluster or hardecodes configuration values in the policy definitions.
```
Comment (mprahl) [inline governance/custom_template.adoc]:
```suggestion
{product-title-short} also supports hub cluster templates to define configuration policies that are dynamically customized to the target cluster. This reduces the need to create separate policies for each target cluster or hardcode configuration values in the policy definitions. 
```

File: governance/custom_template.adoc [hunk 3]
```diff
@@ -361,3 +366,58 @@ spec:
         {{ (fromConfigMap "site-config" "site1" "enabled")  | toBool }}
 ----
 
+[#hub-templates]
+== Support for hub cluster templates in configuration policies
+
+{product-title-short} also supports hub cluster templates to define configuration policies that are dynamically customized to the target cluster. This prevents the creation of separate policies for each target cluster or hardecodes configuration values in the policy definitions. 
+
+Hub cluster templates are based on Golang text template specifications, and uses the `{{hub … hub}}` delimiter for hub cluster templates.
```
Comment (mprahl) [inline governance/custom_template.adoc]:
```suggestion
Hub cluster templates are based on Golang text template specifications, and the `{{hub … hub}}` delimiter indicates a hub cluster template in a configuration policy.
```

File: governance/custom_template.adoc [hunk 4]
```diff
@@ -361,3 +366,58 @@ spec:
...
+| Delimiter
+| {{hub … hub}}
+| {{ … }}
+
+| Functions
+| Set of generic Kubernetes resource such as `lookup` resource, specific `lookup` resources that can be utilized. See <<template-functions,Template functions>> for more information.
+| Set of generic Kubernetes resource such as `lookup` resource, specific `lookup` resources that can be utilized. See <<template-functions,Template functions>> for more information.
+
+| Context
+| A `ManagedClusterName` variable is available, which at runtime, resolves to the name of the target cluster where the policy is propagated.
```
Comment (mprahl) [inline governance/custom_template.adoc]:
```suggestion
| A `.ManagedClusterName` variable is available, which at runtime, resolves to the name of the target cluster where the policy is propagated.
```

File: governance/custom_template.adoc [hunk 5]
```diff
@@ -361,3 +366,58 @@ spec:
...
+| Processing occurs at runtime on the hub cluster during propagation of replicated policies to clusters. Policies and the hub cluster templates within the policies are processed on the hub cluster only when templates are created or updated.
+| Processing occurs in the `ConfigPolicyController` on the managed cluster. Policies are processed periodically, which automatically updates the resolved object definition with data in the referenced resources.
+
+| Access control
+| You can only reference Kuberenetes resources that are in the same namespace as the `Policy` resource.
+| You can reference any resource on the cluster.
+
+| Processing errors
+| Errors from the hub cluster templates are displayed as violations on the specific target cluster that is propagated.
+| Errors from the managed cluster templates are displayed as violations on the specific target cluster where the violationoccurred.
```
Comment (mprahl) [inline governance/custom_template.adoc]:
```suggestion
| Errors from the managed cluster templates are displayed as violations on the specific target cluster where the violation occurred.
```

File: governance/custom_template.adoc [hunk 6]
```diff
@@ -361,3 +366,58 @@ spec:
...
+| Processing
+| Processing occurs at runtime on the hub cluster during propagation of replicated policies to clusters. Policies and the hub cluster templates within the policies are processed on the hub cluster only when templates are created or updated.
+| Processing occurs in the `ConfigPolicyController` on the managed cluster. Policies are processed periodically, which automatically updates the resolved object definition with data in the referenced resources.
+
+| Access control
+| You can only reference Kuberenetes resources that are in the same namespace as the `Policy` resource.
+| You can reference any resource on the cluster.
+
+| Processing errors
+| Errors from the hub cluster templates are displayed as violations on the specific target cluster that is propagated.
```
Comment (mprahl) [inline governance/custom_template.adoc]:
```suggestion
| Errors from the hub cluster templates are displayed as violations on the managed clusters the policy applies to.
```

File: governance/custom_template.adoc [hunk 7]
```diff
@@ -361,3 +366,58 @@ spec:
...
+| Syntax
+| Golang text template specification
+| Golang text template specification
+
+| Delimiter
+| {{hub … hub}}
+| {{ … }}
+
+| Functions
+| Set of generic Kubernetes resource such as `lookup` resource, specific `lookup` resources that can be utilized. See <<template-functions,Template functions>> for more information.
```
Comment (mprahl) [inline governance/custom_template.adoc]:
```suggestion
| Set of generic Kubernetes resource such as `lookup` resource, specific `lookup` resources that can be utilized. See <<template-functions,Template functions>> for more information. Resources referenced in the `lookup` function must be in the same namespace as the policy due to a {product-short-title} security restriction. Due to this, the namespace can be set to an empty string when using these functions.
```

**Comment by dockerymick** [comment]
@mprahl Thanks for your patience Matt! I will complete this after the lunch hour

**Comment by mprahl** [comment]
> @mprahl Thanks for your patience Matt! I will complete this after the lunch hour

No problem @dockerymick! I need to make an additional change. I'll let you know when it's ready for review again.

Thanks for your help!

**Comment by mprahl** [comment]
@dockerymick okay, I made a few commits with some changes I wanted but I did not address the comments I previously left.

It's ready for review now. Thanks!

File: governance/custom_template.adoc (hunk 5 above)
Comment (dockerymick) [reply governance/custom_template.adoc]: Great catch, thank you ! Accepting suggestion

**Comment by dockerymick** [comment]
@mprahl thank you for your detailed review and suggestions! I made a couple of changes and accepted some suggestions. Let me know you have questions. @ckandag may you also review when you have an opportunity? 

May you both take a look for approval

**Review by mprahl — APPROVED** [review]
Awesome job!

**Pull request merged**